pkg/engine/internal/executor/dataobjscan.go

```diff
@@ -386,4 +388,45 @@ func (s *dataobjScan) Close() {
 	s.streams = nil
 	s.streamsInjector = nil
 	s.reader = nil
+	s.scope = nil
+}
+
+// recordReaderStats records statistics from the [logs.Reader] to the xcap scope.
+func (s *dataobjScan) recordReaderStats() {
```

Comment:
nit: should there be a comment here saying we can remove this once we add xcap support into the dataobj package?